packages/@glimmer/manager/lib/internal/defaults.ts

```diff
@@ -0,0 +1,42 @@
+import { buildCapabilities } from '../util/capabilities';
+
+import type { CapturedArguments as Arguments, HelperCapabilities } from '@glimmer/interfaces';
+
+type FnArgs<Args extends Arguments = Arguments> =
+  | [...Args['positional'], Args['named']]
+  | [...Args['positional']];
+
+interface FunctionHelperState<Args extends Arguments = Arguments> {
+  fn: <Return>(...args: FnArgs<Args>) => Return;
+  args: Args;
+}
+
+export class FunctionHelperManager<State extends FunctionHelperState> {
+  capabilities = buildCapabilities({
+    hasValue: true,
+    hasDestroyable: false,
+    hasScheduledEffect: false,
+  }) as HelperCapabilities;
+
+  createHelper(fn: State['fn'], args: State['args']) {
```

Comment:
```suggestion
  createHelper(fn: State['fn'], args: State['args']): State {
```

Reply:
(otherwise I feel these changes are kinda superficial)

Reply:
Hm, there may be a disconnect, it's not about not using interfaces, I am commenting on using explicit return types.

Reply:
> interface
> inference

these words are annoyingly similar lol

Reply:
there is an eslint-todo system out there somewhere, yeah? been meaning to look into one for work.

Reply:
but anywho, if you're open to a lint, I can figure that out, too, and just add these changes

Reply:
lol sorry about that. Yeah, I don't think we are "against inference" in general (e.g. it's fine to use them when declaring variables), but definitely not in return positions, since they are part of the function/method's API (public or private), so you don't want a refactor in the function body to _accidentally_ change the API of the function. It's fine if you did it intentionally and had to change the return type, but that would be a good thing to see in the diff during a PR review (for example).

So I think roughly the rule is don't use inference for things that affects your consumers, and use them all you want inside your private implementation details. Mostly that applies to return types but probably a good practice for things like exported `const`s, etc.

Arguably that should probably apply for function parameters too (when defaulting), but usually we seem ok with it, probably because it would be pretty obvious in the PR when you made a change in the parameter list, whereas the return type can change due to any changes in the function body which is much less obvious.